src/dhali/transaction_utils.py

```diff
@@ -272,6 +337,66 @@ async def update_estimated_cost_with_exact(
 
 
 
+
+async def validate_exact_claim(
+    claim, estimated_claim_uuid: str, single_request_exact_cost: int, db
+) -> float:
+    """
+    TODO
+
+    Parameters
+    ----------
+    client : xrpl.clients.JsonRpcClient
+    claim : str
+    single_request_cost_estimate : int
+    exact_cost : int
+
+    :raises:
+        HTTPException: 402 status code HTTPException
+
+    Returns
+    -------
+    void
+    """
+    try:
+        parsed_claim = json.loads(claim)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=402,
+            detail=f"You must provide a payment channel claim that can be parsed via json.loads. Error: {e}.",
+        )
+    
+    uuid_channel_id = str(uuid.uuid5(uuid.NAMESPACE_URL, parsed_claim["channel_id"]))
+    estimated_payment_claim_doc_ref = db.collection(root_private_collection_name).document(uuid_channel_id).collection(estimate_collection_name).document(estimated_claim_uuid)
+    exact_payment_claim_doc_ref = db.collection(root_private_collection_name).document(uuid_channel_id).collection(exact_collection_name).document(estimated_claim_uuid)
+    estimated_payment_claim_doc = estimated_payment_claim_doc_ref.get()
+
+    if estimated_payment_claim_doc.get("authorized_to_claim") != parsed_claim["authorized_to_claim"]:
+        raise HTTPException(
+            status_code=402,
+        )
+    
+    if estimated_payment_claim_doc.get("payment_claim") != claim:
+        raise HTTPException(
+            status_code=402,
```

Comment:
402 is for payment required iirc. Could you please explain why we would return it here?

Reply:
I think both of these should be 500 errors